core/lib/via_btc_client/src/indexer/parser.rs

```diff
@@ -0,0 +1,321 @@
+                .try_into()
+                .ok()?,
+        ));
+        let da_identifier = std::str::from_utf8(instructions.get(4)?.push_bytes()?.as_bytes())
+            .ok()?
+            .to_string();
+        let blob_id = std::str::from_utf8(instructions.get(5)?.push_bytes()?.as_bytes())
+            .ok()?
+            .to_string();
+
+        Some(Message::L1BatchDAReference(L1BatchDAReference {
+            common: common_fields.clone(),
+            input: L1BatchDAReferenceInput {
+                l1_batch_hash,
+                l1_batch_index,
+                da_identifier,
+                blob_id,
+            },
+        }))
+    }
+
+    fn parse_proof_da_reference(
+        &self,
+        instructions: &[Instruction],
+        common_fields: &CommonFields,
+    ) -> Option<Message> {
+        if instructions.len() < 5 {
+            return None;
+        }
+
+        let l1_batch_reveal_txid =
+            Txid::from_slice(instructions.get(2)?.push_bytes()?.as_bytes()).ok()?;
+        let da_identifier = std::str::from_utf8(instructions.get(3)?.push_bytes()?.as_bytes())
+            .ok()?
+            .to_string();
+        let blob_id = std::str::from_utf8(instructions.get(4)?.push_bytes()?.as_bytes())
+            .ok()?
+            .to_string();
+
+        Some(Message::ProofDAReference(ProofDAReference {
+            common: common_fields.clone(),
+            input: ProofDAReferenceInput {
+                l1_batch_reveal_txid,
+                da_identifier,
+                blob_id,
+            },
+        }))
+    }
+
+    fn parse_l1_to_l2_message(
+        &self,
+        tx: &Transaction,
+        instructions: &[Instruction],
+        common_fields: &CommonFields,
+    ) -> Option<Message> {
+        if instructions.len() < 5 {
+            return None;
+        }
+
+        let receiver_l2_address =
+            EVMAddress::from_slice(instructions.get(2)?.push_bytes()?.as_bytes());
+        let l2_contract_address =
+            EVMAddress::from_slice(instructions.get(3)?.push_bytes()?.as_bytes());
+        let call_data = instructions.get(4)?.push_bytes()?.as_bytes().to_vec();
+
+        let amount = tx
+            .output
+            .iter()
+            .find(|output| output.script_pubkey.is_p2wpkh())
+            .map(|output| output.value)
+            .unwrap_or(Amount::ZERO);
+
+        Some(Message::L1ToL2Message(L1ToL2Message {
+            common: common_fields.clone(),
+            amount,
+            input: L1ToL2MessageInput {
+                receiver_l2_address,
+                l2_contract_address,
+                call_data,
+            },
+        }))
+    }
+}
+
+fn is_via_inscription_protocol(instructions: &[Instruction]) -> Option<usize> {
+    // TODO: also check first part of the script (OP_CHECKSIG and other stuff)
+    instructions.iter().position(|instr| {
+        matches!(instr, Instruction::PushBytes(bytes) if bytes.as_bytes() == b"Str('via_inscription_protocol')")
```

Comment:
Are you sure it is serialized with `Str(...)`, check here https://github.com/vianetwork/via-server/blob/5644fb3e73939190f31e3d91f9f6ccaab053c54c/core/lib/via_btc_client/src/inscriber/script_builder.rs#L14?

Reply:
fixed